src/commands/deploy/utils.ts

```diff
@@ -207,7 +212,7 @@ export async function uploadEnvVarsFromFile(
 
     // This is best effort, we should encourage the user to use `--env <envFile>` to set the correct env file path.
     // Search for possible .env files in the project directory and use the first
-    const envFile = await findAnEnvFile(cwd);
+    const envFile = envPath ? path.join(envPath, ".env") : await findAnEnvFile(cwd);
```

Comment:
I think `envPath` is the path that points directly to the env file, there is no need for `path.join`